tools/image-audit/identity/identityregistry.js

```diff
@@ -0,0 +1,138 @@
+    const identityRegistry = IdentityRegistry.#identityRegistry;
+
+    if (Object.prototype.isPrototypeOf
+      .call(AbstractIdentity.prototype, registeredClass.prototype)) {
+      // Identity classes
+      if (registeredClass.identifyPreflight) {
+        identityRegistry.#identifyPreflightSet.add(registeredClass);
+      }
+      if (registeredClass.identifyPostflight) {
+        identityRegistry.#identifyPostflightSet.add(registeredClass);
+      }
+      if (registeredClass.identifyPostflightWithCanvas) {
+        identityRegistry.#identifyPostflightWithCanvasSet.add(registeredClass);
+      }
+      if (registeredClass.identifyPostError) {
+        identityRegistry.#identifyPostErrorSet.add(registeredClass);
+      }
+      if (registeredClass.type) {
+        identityRegistry.#types.push(registeredClass.type);
+      }
+      identityRegistry.#registeredIdentityClasses.add(registeredClass);
+    } else if (Object.prototype.isPrototypeOf
+      .call(AbstractIdentityHashProvider.prototype, registeredClass.prototype)) {
+      // Hash classes
+      if (!identityRegistry.#identityHashProvider) {
+        identityRegistry.#identityHashProvider = registeredClass;
+      } else if (identityRegistry.#identityHashProvider.providerPriority
+        > registeredClass.providerPriority) {
+        identityRegistry.#identityHashProvider = registeredClass;
+      }
+    } else if (Object.prototype.isPrototypeOf
+      .call(AbstractCacheProvider.prototype, registeredClass.prototype)) {
+      // Cache classes
+      if (!identityRegistry.#identityCache) {
+        // eslint-disable-next-line new-cap
+        identityRegistry.#identityCache = new registeredClass();
+      } else if (identityRegistry.#identityCache.providerPriority
+        > registeredClass.providerPriority) {
+        // eslint-disable-next-line new-cap
+        identityRegistry.#identityCache = new registeredClass();
+      }
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async #runIdentifications(set, methodName, identityValues, identityState) {
+    const promises = [];
+    const selectedIdentifiers = Array.from(set)
+      .filter((clazz) => identityValues.selectedIdentifiers.has(clazz.type));
+
+    // cut down the identityState to just a piece for the identity itself.
+    selectedIdentifiers.forEach((clazz) => {
+      if (!identityState[clazz.type]) {
+        identityState[clazz.type] = {};
+      }
+
+      promises.push(clazz[methodName](identityValues, identityState[clazz.type]));
+    });
+
+    const results = await Promise.allSettled(promises);
+    results
+      .filter((result) => result.status === 'rejected')
+      // eslint-disable-next-line no-console
+      .forEach((error) => console.error('Error handling identification', error));
+  }
+
+  async identifyPreflight(identityValues, identityState) {
+    await this.#runIdentifications(this.#identifyPreflightSet, 'identifyPreflight', identityValues, identityState);
+  }
```

Comment:
technically, you don't need the `await` on the last method in a function and could just return the statement, or even just leave the statement without the return as async function will return it by default. The consumer can decide whether to await or not as required

Reply:
good point. Will change to return